tests/actions/test_action_typedicts.py

```
@@ -0,0 +1,234 @@
+            )
+
+    if errors:
+        assert False, "Field mismatches between actions and TypedDicts:\n  - " + "\n  - ".join(errors)
+
+
+def test_typeddict_type_field_matches_action_type() -> None:
+    """Test that each TypedDict's 'type' field Literal matches the action's type."""
+    errors: list[str] = []
+    mapping = _get_action_to_typeddict_mapping()
+
+    for action_type, (action_cls, typeddict_cls) in mapping.items():
```

Ruff (0.14.8)

Loop control variable `action_cls` not used within loop body

Rename unused `action_cls` to `_action_cls`

(B007)